pandas/core/arrays/categorical.py

```diff
@@ -2133,7 +2134,7 @@ def _concat_same_type(self, to_concat):
 
     # ------------------------------------------------------------------
 
-    def is_dtype_equal(self, other):
+    def categories_match_up_to_permutation(self, other: "Categorical") -> bool:
```

Comment:
umm, why this name? what is wrong with the existing one?

Reply:
> what is wrong with the existing one?

`is_dtype_equal(self, other)` means something different from `self.is_dtype_equal(other)` which i find confusing

>  why this name? 

it accurately describes what is being checked

Reply:
i don't have a problem with this as a private method, and its ok to deprecate the .is_dtype_equal but we shouldn't offer this as a replacement.

Reply:
privatized+green